backend/main.py

```diff
@@ -10,6 +12,50 @@
     env.reset(sys.argv[1])
     visualiser = Visualiser(env.realm)
 
+    #Temporary constants
+    car_size = 2
+    
     while True:
-        env.step({})
+        actions = {}
+        #Work out actions
+        for actor in env.realm.actors.values():
+            if type(actor) is Depot:
+                for t in actor._storage:
+                    truck = actor._storage[t]
+                    if truck.done == False: #Truck is waiting to be released
+                        next_road = env.realm.edges[truck.next_container_id]
+                        if type(next_road) is Road:
+                            first_car_pos = next_road._trucks[0].position * next_road._length
+                            if first_car_pos > car_size: #There is space on the road
+                                #Release this truck
+                                actions[actor.id] = float(truck.id)
+
+            elif type(actor) is Road:
+                for x in range(len(actor._trucks)):
+                    i = len(actor._trucks) - x - 1
```

Comment:
You can iterate over this backwards with `for i in range(len(actor._trucks)-1, -1, -1)`